tools/zulint/custom_rules.py

```diff
@@ -15,183 +15,212 @@
     RuleList = List[Dict[str, Any]]
     LineTup = Tuple[int, str, str, str]
 
-def get_line_info_from_file(fn):
-    # type: (str) -> List[LineTup]
-    line_tups = []
-    for i, line in enumerate(open(fn)):
-        line_newline_stripped = line.strip('\n')
-        line_fully_stripped = line_newline_stripped.strip()
-        if line_fully_stripped.endswith('  # nolint'):
-            continue
-        tup = (i, line, line_newline_stripped, line_fully_stripped)
-        line_tups.append(tup)
-    return line_tups
-
-def get_rules_applying_to_fn(fn, rules):
-    # type: (str, RuleList) -> RuleList
-    rules_to_apply = []
-    for rule in rules:
-        excluded = False
-        for item in rule.get('exclude', set()):
-            if fn.startswith(item):
-                excluded = True
-                break
-        if excluded:
-            continue
-        if rule.get("include_only"):
-            found = False
-            for item in rule.get("include_only", set()):
-                if item in fn:
-                    found = True
-            if not found:
+
+class RuleList:
```

Comment:
Is this close to what you wanted? Just wanted to confirm before investing too much into the idea.

Reply:
I thoughts I'd already posted a comment on this, but don't see it here, so maybe GitHub lost it.  Yes, this is the concept I had in mind.

Reply:
Cool! I will proceed in this direction then.